src/zodbverify/verify_oid.py

```diff
@@ -13,6 +20,173 @@
 logger = logging.getLogger("zodbverify")
 
 
+class Refbuilder(object):
+
+    def __init__(self, storage, connection):
+        self.storage = storage
+        self.connection = connection
+        self.seen = []
+        self.refs = defaultdict(list)
+        self.msg = []
+        self.stop_recurse = [
+            '0x00',  # persistent.mapping.PersistentMapping
+            '0x01',  # OFS.Application.Application
+            '0x11',  # Products.CMFPlone.Portal.PloneSite
+        ]
+
+    def get_oid_report(self, oid, level=0, max_level=600, verbose=False):
+        self.msg.append('The oid {} is referenced by:\n'.format(oid))
+        self.inspect_reference_tree(oid, level=0, max_level=600, verbose=verbose)
+        logger.info('\n'.join(self.msg))
+
+    def _build_ref_tree(self):
+        logger.info('Building a reference-tree of ZODB...')
+        count = 0
+        next_ = None
+        while True:
+            count += 1
+            oid, tid, data, next_ = self.storage.record_iternext(next_)
+
+            # For each oid create a list of oids that reference it
+            # Can be used for reverse lookup similar to what fsoids.py
+            # in ZODB does.
+            oid_refs = get_refs(data)
+            if oid_refs:
+                for referenced_oid, class_info in oid_refs:
+                    self.refs[oid_repr(referenced_oid)].append(oid_repr(oid))
```

Comment:
Why do you use the `oid_repr` as key? IMO this is overhead. The `oid` should be hashable as well, has a smaller memory footprint and lots of CPU cycles would be saved as well.